### DISCUSSION TITLE
Want to understand Web3Auth integration with firebase authentication is possible currently?

### DISCUSSION THREAD

**Discussion by 1swaraj** [Web3Auth/Web3Auth#321] [Help]
Hi guys, want to understand Web3Auth integration with firebase authentication is possible currently?

However, as most of our server side assets are protected with firebase auth, so we're looking to extend the firebase provider.

**Comment by yashovardhan** [comment]
Yes, Web3Auth can be integrated with Firebase Authentication. With our custom authentication system, you can pass your firebase `id_token` to web3auth and generate a private key from Web3Auth. Checkout our SDK Reference around Custom Authentication to know more: https://web3auth.io/docs/sdk/web/core/custom-authentication

Additionally, you can checkout our demos for Web3Auth + Firebase integration.
- https://github.com/Web3Auth/Web3Auth/tree/master/demo/firebase-next-app
- https://github.com/Web3Auth/Web3Auth/tree/master/demo/firebase-react-app

We'll be soon releasing a proper end to end guide around this integration as well. We will update this discussion accordingly once that happens.

**Reply by shahbaz17** [reply]
Guide is live https://web3auth.io/docs/guides/firebase
